gateway/ln-gateway/src/client.rs

```diff
@@ -57,49 +77,68 @@ impl GatewayClientBuilder {
             mint_channel_id,
             gateway: gateway.clone(),
         });
-        registry.attach(GatewayClientInitV2 { gateway });
-
-        let db_path = self.work_dir.join(format!("{federation_id}.db"));
-
-        let rocksdb = fedimint_rocksdb::RocksDb::open(db_path.clone()).map_err(|e| {
-            GatewayError::DatabaseError(anyhow::anyhow!("Error opening rocksdb: {e:?}"))
-        })?;
-        let db = Database::new(rocksdb, ModuleDecoderRegistry::default());
+        registry.attach(GatewayClientInitV2 {
+            gateway: gateway.clone(),
+        });
 
         let mut client_builder = Client::builder(db)
             .await
             .map_err(GatewayError::DatabaseError)?;
         client_builder.with_module_inits(registry);
         client_builder.with_primary_module(self.primary_module);
         client_builder.with_connector(connector);
+        Ok(client_builder)
+    }
 
-        let client_secret = if let Ok(secret) =
-            Client::load_decodable_client_secret::<[u8; 64]>(client_builder.db_no_decoders()).await
-        {
-            secret
+    pub async fn build(
+        &self,
+        config: FederationConfig,
+        gateway: Arc<Gateway>,
+        mnemonic: &Mnemonic,
+    ) -> Result<fedimint_client::ClientHandleArc> {
+        let invite_code = config.invite_code.clone();
+        let federation_id = invite_code.federation_id();
+        let db_path = self.work_dir.join(format!("{federation_id}.db"));
+
+        let (db, root_secret) = if db_path.exists() {
+            let rocksdb = fedimint_rocksdb::RocksDb::open(db_path.clone()).map_err(|e| {
+                GatewayError::DatabaseError(anyhow::anyhow!("Error opening rocksdb: {e:?}"))
+            })?;
+            let db = Database::new(rocksdb, ModuleDecoderRegistry::default());
+            let root_secret = self.client_plainrootsecret(&db).await?;
+            (db, root_secret)
         } else {
-            info!("Generating secret and writing to client storage");
-            let secret = PlainRootSecretStrategy::random(&mut thread_rng());
-            Client::store_encodable_client_secret(client_builder.db_no_decoders(), secret)
-                .await
-                .map_err(GatewayError::ClientStateMachineError)?;
-            secret
+            let db = gateway
+                .gateway_db
+                .with_prefix(federation_id.to_prefix().to_bytes());
+            let root_secret = Bip39RootSecretStrategy::<12>::to_root_secret(mnemonic);
+            (db, root_secret)
```

Comment:
We directly return the root secret here, but should be deriving a per-federation secret as described in the [secret derivation docs](https://github.com/fedimint/fedimint/blob/master/docs/secret_derivation.md).

The current version isn't *horribly* broken because out of abundance of caution clients internally derive a key based on the federation id, but to be compatible with other wallet tools using the standard derivation path would be nice.

Reply:
Hmm interesting, in early discussions with @dpc I thought the client was already deriving a per-federation secret using the federation id, and that adding it here would actually be doing the derivation twice unnecessarily. I actually originally had it like that, but ended up removing the derivation after our discussion. 

Is it recommended to derive a per-federation secret here?

Reply:
We are doing the derivation twice if following the standard, so if an integrator doesn't follow the standard then they don't shoot themselves (or rather their users) in the foot. 

The reasoning was: 
1. It doesn't degrade security 
2. The time it takes is negligible
3. Integrators won't know about the second client-internal derivation, so it won't feel weird to them

Reply:
Fixed